android-auto-app/src/main/java/com/mapbox/navigation/examples/androidauto/car/MainCarSession.kt

```diff
@@ -29,140 +27,112 @@ import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
 import com.mapbox.navigation.core.lifecycle.requireMapboxNavigation
 import com.mapbox.navigation.core.trip.session.TripSessionState
 import com.mapbox.navigation.examples.androidauto.CarAppSyncComponent
+import com.mapbox.navigation.examples.androidauto.ReplayRouteTripSession
 import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 
-@OptIn(MapboxExperimental::class, ExperimentalPreviewMapboxNavigationAPI::class)
+@OptIn(MapboxExperimental::class)
 class MainCarSession : Session() {
 
-    private var mainCarContext: MainCarContext? = null
-    private lateinit var mainScreenManager: MainScreenManager
-    private lateinit var mapboxCarMap: MapboxCarMap
-    private lateinit var navigationManager: MapboxCarNavigationManager
-    private val replayRouteTripSession = ReplayRouteTripSession()
-    private val mainCarMapLoader = MainCarMapLoader()
+    private val mapboxCarMapLoader = MapboxCarMapLoader()
+    private val mapboxCarMap = MapboxCarMap().registerObserver(mapboxCarMapLoader)
+    private val mapboxCarContext = MapboxCarContext(lifecycle, mapboxCarMap).prepareScreens()
+
+    @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
     private val mapboxNavigation by requireMapboxNavigation()
 
     init {
-        MapboxNavigationApp.attach(this)
+        // Decide how you want the car and app to interact. In this example, the car and app
+        // are kept in sync where they essentially mirror each other.
         CarAppSyncComponent.getInstance().setCarSession(this)
 
+        // Add BitmapWidgets to the map that will be shown whenever the map is visible.
         val logoSurfaceRenderer = CarLogoSurfaceRenderer()
         val compassSurfaceRenderer = CarCompassSurfaceRenderer()
-        logAndroidAuto("MainCarSession constructor")
         lifecycle.addObserver(object : DefaultLifecycleObserver {
-
             override fun onCreate(owner: LifecycleOwner) {
-                logAndroidAuto("MainCarSession onCreate")
-                val mapInitOptions = MapInitOptions(
-                    context = carContext,
-                    styleUri = mainCarMapLoader.mapStyleUri(carContext.isDarkMode)
-                )
-                mapboxCarMap = MapboxCarMap(mapInitOptions)
-                mainCarContext = MainCarContext(carContext, mapboxCarMap)
-                mainScreenManager = MainScreenManager(mainCarContext!!)
-                navigationManager = MapboxCarNavigationManager(carContext)
-                observeScreenManager()
+                // You must give the MapboxCarMap an instance of the carContext.
+                mapboxCarMap.setup(carContext, MapInitOptions(context = carContext))
```

Comment:
Why don't you use `MapboxCarMapSessionInstaller`?

Reply:
I wanted to use the examples repo for non-experimental apis to start. the `qa-test-app` and `1tap-android` are used for adopting the experimental apis.

It's likely that we make `MapboxCarMapSessionInstaller` permanent. But I like the `MapboxCarContextOwner` and I'd consider moving that up to `MapboxCarMap` so that the `MapboxCarMap.setup` function could be removed. The issue is still, giving `MapInitOptions` to `MapboxCarMap`.

Nothing wrong with `MapboxCarMapSessionInstaller` though.. I'm not actively working on replacing it so we could just go ahead with using it.